File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n, a int
	var s string
	var max float64
	fmt.Fscanf(reader, "%d\n", &n)
	var graph = [][]float64{}
	for i := 0; i < n; i++ {
		s, _ = reader.ReadString('\n')
		s = strings.TrimSpace(s)
		strArr := strings.Split(s, " ")
		var tmp = []float64{}
		for j := 0; j < len(strArr); j++ {
			a, _ = strconv.Atoi(strArr[j])
			tmp = append(tmp, float64(a))
		}
		graph = append(graph, tmp)
	}
	var res = graph
	for i := 1; i < n; i++ {
		for j := 0; j < i+1; j++ {
			if j == 0 {
				res[i][j] = res[i-1][j] + graph[i][j]
			} else if j == i {
				res[i][j] = res[i-1][j-1] + graph[i][j]
			} else {
				res[i][j] = math.Max(res[i-1][j-1], res[i-1][j]) + graph[i][j]
			}
		}
	}
	for _, v := range res[len(res)-1] {
		max = math.Max(v, max)
	}
	fmt.Printf("%0.f", max)
}

File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
graph = []
for i in range(n):
    graph.append(list(map(int, input().split())))
res = [[0] * i for i in range(1, n+1)]
res[0][0] = graph[0][0]
for i in range(1, n):
    for j in range(i+1):
        if j == 0:
            res[i][j] = res[i-1][j] + graph[i][j]
        elif i == j:
            res[i][j] = res[i-1][j-1] + graph[i][j]
        else:
            res[i][j] = max(res[i - 1][j - 1], res[i - 1][j]) + graph[i][j]
print(max(res[-1]))